damnit/gui/main_window.py

```diff
@@ -50,6 +50,11 @@ class MainWindow(QtWidgets.QMainWindow):
 
     context_dir_changed = QtCore.pyqtSignal(str)
     save_context_finished = QtCore.pyqtSignal(bool)  # True if saved
+    context_saved = QtCore.pyqtSignal()
```

Comment:
> AFAICT the save_context_finished signal is doing the same thing as context_saved

Ah, that's a mistake on my part. When you ask the editor to save, it only saves it if there are no major errors, and the idea was that `context_saved` fires when it has actually saved (parsing succeeded), but `save_context_finished` fires when the pre-save check finishes, whether or not it actually saves.

> I saw that the only place where save_context_finished is used is in the tests. Are you aware if this was added just for testing purposes

Essentially, yes. It could be used in the application code in the future, but it's not now. I needed some way for the tests to hook in to when the saving had finished, successfully or not.  I agree adding a new signal isn't great, but I didn't think of a better way.

Reply:
Ah, ok. That clarifies my doubts.

Do you think that keeping only one signal and filtering the calls with `saving == true` on the receiving side could be better or do you prefer keeping the signals separated to allow independent evolution of the two?

Reply:
No strong preference - do you prefer to have one signal?

Reply:
I think that would be a little bit cleaner but is not something that I would deem blocking or very important. Just to give you some context when I saw it my first impression was that the other signal was some leftover of old function that was removed and was forgotten there. So I don't see this as a problem other than that it could be puzzling for someone reading the code (probably not many people :laughing:).

Reply:
Fair enough, I've got rid of the `context_saved` signal and used `save_context_finished` instead.